app/components/Swapbox.tsx

```diff
@@ -362,31 +360,25 @@ export const Swapbox = ({ fixedProductMarketMaker }: SwapboxProps) => {
               </div>
             )}
           </div>
-          {isDisconnected ? (
-            <ConnectButton width="full" size="lg">
-              Connect
-            </ConnectButton>
-          ) : unsupportedNetwork ? (
-            <Button width="full" variant="pastel" size="lg" disabled>
-              Unsupported network
-            </Button>
-          ) : !tokenAmountIn || +tokenAmountIn === 0 ? (
-            <Button width="full" variant="pastel" size="lg" disabled>
-              Enter amount
-            </Button>
-          ) : currentState.isLoading ? (
-            <Button width="full" variant="pastel" size="lg" disabled>
-              Fetching price
-            </Button>
-          ) : +tokenAmountIn > +formatEther(currentState.balance) ? (
-            <Button width="full" variant="pastel" size="lg" disabled>
-              Insufficient {currentState.inToken.symbol} balance
-            </Button>
-          ) : (
-            <Button width="full" variant="pastel" size="lg" onClick={openBetModal}>
-              {currentState.buttonText}
-            </Button>
-          )}
+          <AccountStateButton>
+            {!tokenAmountIn || +tokenAmountIn === 0 ? (
+              <Button width="full" variant="pastel" size="lg" disabled>
+                Enter amount
+              </Button>
+            ) : currentState.isLoading ? (
+              <Button width="full" variant="pastel" size="lg" disabled>
+                Fetching price
+              </Button>
+            ) : +tokenAmountIn > +formatEther(currentState.balance) ? (
+              <Button width="full" variant="pastel" size="lg" disabled>
+                Insufficient {currentState.inToken.symbol} balance
+              </Button>
+            ) : (
+              <Button width="full" variant="pastel" size="lg" onClick={openBetModal}>
+                {currentState.buttonText}
+              </Button>
```

Comment:
We can also abstract the button style props and receive `onClick` and `disabled` props. What do you think?

Reply:
I think the whole structure can be abstracted, that's why I did it that way because what changes are the balances, the amounts, the "final" button label, and the final button `onClick` function. The only actionable buttons are the "final" and the Connect buttons.

For this specific suggestion, if we go this way I'd abstract the `onClick` and `variant` props + the `children` (the button label) and move this Button inside the AccountStateButton. WDYT?

Reply:
I wouldn't abstract everything. For example, current logic inside the `children` would be the same but only without the button element. I would let the developer define things like it was a normal button. This way it would be more flexible.

I'm suggesting this because it's the only thing in common with all the Button elements. Style, disable and onClick props are used "everywhere" (disabled and onClick are omitted and some buttons). Otherwise, the logic outside this is different for each use case.